src/rate-limit-view/rate-limit-view.ts

```diff
@@ -0,0 +1,87 @@
+import COLORS from "../data/rate-limit-view-colors.json";
+
+interface RateLimitColors {
+	negative: string,
+	positive: string[]
+}
+
+const RATE_LIMIT_COLORS: RateLimitColors = COLORS;
+const RATE_LIMIT = 333;
+const COLOR_INTERVALL = 20;
+const COLOR_MAX_MS = RATE_LIMIT_COLORS.positive.length * COLOR_INTERVALL;
+
+export class RateLimitView {
+	constructor(Elements: any, Options: any) {
+		Elements.$body.attr("data-RateLimitView", "Disabled");
+		$("<span id=ratelim class=ratelimit>RL: <span id=rate></span></span>").insertBefore(".CANT_REPLY");
+		$(".ratelimit").css({
+			background: "transparent",
+			display: "none",
+			float: "right",
+			"font-size": "smaller",
+			"margin-left": "10px",
+			"margin-top": "5px",
+		});
+		this.initOptions(Elements, Options);
+	}
+
+	initOptions(Elements: any, Options: any): void {
+		let oldtime = Date.now();
+		let newtime = Date.now();
+		let bgColor = "";
+		let fontColor = "";
+		let isDelta = false;
+		let isEnabled = false;
+		Options.addSelect({
+			default: 0,
+			help: "Rate limit viewer by post. Not real, just estimate lol",
+			label: "RATE LIMIT VIEW",
+			onchange: function() {
+				const val = this.val();
+				Elements.$body.attr("data-RateLimitView", val);
+
+				if(val == "Normal" || val == "Delta") {
+					$(".ratelimit").css("display", "initial");
+					$(".help-toggle").css("display", "none");
+					isEnabled = true;
+				}
+				if(val == "Disabled") {
+					$(".ratelimit").css("display", "none");
+					$(".help-toggle").css("display", "initial");
+					$(".bottom-area").css("background", "transparent");
+					$(".bottom-area").css("color", "");
+					isEnabled = false;
+				}
+				isDelta = val == "Delta";
+			},
```

Comment:
This could potentially be a class method. `this` should work the same even from within a method, though generally confusing uses of `this` should be avoided.

Reply:
this.val() is a function of the object passed into Options.addSelect

I've had to work around this in #32 by aliasing "this"

Reply:
This will work even with a class method, especially since you don't need to access the `this` referencing `RateLimitView`:

```java
class Foo {
	log() {
		console.log(this);
	}
}

class Bar {
	test() {
		new Foo().log.call(this);
	}
}

new Bar().test();
```

Reply:
Does 7b2d99f resolve this?

Reply:
> since you don't need to access the `this` referencing `RateLimitView`

If I move it to a class method I do though, or am I missing something?